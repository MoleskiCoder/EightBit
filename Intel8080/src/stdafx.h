#ifdef _MSC_VER
#pragma once
#endif

#include <cstdint>
#include <functional>

#include <iostream>
#include <fstream>

#include <string>
#include <array>
#include <vector>

#if defined(_M_X64) || defined(_M_IX86 )
#	define HOST_LITTLE_ENDIAN
#else
#	define HOST_BIG_ENDIAN
#endif
