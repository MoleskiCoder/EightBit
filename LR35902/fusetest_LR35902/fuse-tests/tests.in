00
0000 0000 0000 0000 0000 0000
0 1
0000 00 -1
-1

01
0000 0000 0000 0000 0000 0000
00 1
0000 01 12 34 -1
-1
