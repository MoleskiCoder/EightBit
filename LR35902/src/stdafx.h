#pragma once

#include <cstdint>
#include <stdexcept>

#include <functional>

#include <vector>
#include <array>
#include <map>

#include <bitset>
#include <string>

#include <algorithm>

#include <iostream>
#include <fstream>
#include <sstream>
#include <iomanip>

#include <boost/format.hpp>

#include <SDL.h>

#if SDL_BYTEORDER == SDL_LIL_ENDIAN
#define HOST_LITTLE_ENDIAN
#endif

#if SDL_BYTEORDER == SDL_BIG_ENDIAN
#define HOST_BIG_ENDIAN
#endif

#ifdef _MSC_VER
#pragma comment(lib, "SDL2.lib")
#pragma comment(lib, "SDL2main.lib")
#endif
