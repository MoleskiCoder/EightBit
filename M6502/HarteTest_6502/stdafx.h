#pragma once

#include <cassert>
#include <chrono>
#include <cstdint>
#include <exception>
#include <filesystem>
#include <fstream>
#include <iostream>
#include <map>
#include <memory>
#include <sstream>
#include <stdexcept>
#include <string>
#include <utility>
#include <vector>

#include <Bus.h>
#include <Ram.h>
#include <mos6502.h>
#include <Disassembly.h>
#include <Symbols.h>

//#define TEST_JSON_PERFORMANCE

#define USE_SIMDJSON_JSON	// 13 seconds (19)
//#define USE_RAPIDJSON_JSON	// 19 seconds (22)
//#define USE_BOOST_JSON	// 26 seconds (32)
//#define USE_NLOHMANN_JSON	// 56 seconds (69)
//#define USE_JSONCPP_JSON	// 80 seconds (91)

#ifdef USE_SIMDJSON_JSON
#	define JSON_PREFER_PASS_BY_VALUE
#	define JSON_PREFER_REUSE_OF_PARSER
#	include "simdjson/simdjson.h"
#endif

#ifdef USE_RAPIDJSON_JSON
#	define RAPIDJSON_HAS_STDSTRING	1
#	define RAPIDJSON_SSE42
#	define JSON_INSITU_PARSE
#   include "rapidjson/document.h"
#endif

#ifdef USE_BOOST_JSON
#	include <boost/json.hpp>
#endif

#ifdef USE_NLOHMANN_JSON
#	define JSON_USE_IMPLICIT_CONVERSIONS 0
#	include "nlohmann/json.hpp"
#endif

#ifdef USE_JSONCPP_JSON
#	define JSON_PREFER_REUSE_OF_PARSER
#	include <json/json.h>
#endif
