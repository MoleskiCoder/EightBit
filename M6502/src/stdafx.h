#ifdef _MSC_VER
#pragma once
#endif

#include <array>
#include <cstdint>
#include <fstream>
#include <functional>
#include <iomanip>
#include <map>
#include <sstream>
#include <string>
#include <utility>
#include <vector>

#include <boost/algorithm/string.hpp>
#include <boost/algorithm/string/join.hpp>
#include <boost/algorithm/string/regex.hpp>
#include <boost/regex.hpp>

#include <EightBitCompilerDefinitions.h>
#include <EventArgs.h>
#include <LittleEndianProcessor.h>
#include <Register.h>
#include <Signal.h>