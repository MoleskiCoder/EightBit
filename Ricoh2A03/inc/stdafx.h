#pragma once

#include <mos6502.h>
#include <Bus.h>
