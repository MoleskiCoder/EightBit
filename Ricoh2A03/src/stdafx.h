#pragma once

#include <cstdint>
#include <mos6502.h>
#include <Bus.h>
