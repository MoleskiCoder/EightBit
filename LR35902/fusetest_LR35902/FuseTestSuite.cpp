#include "stdafx.h"
#include "FuseTestSuite.h"
#include "FuseTestRunner.h"

Fuse::TestSuite::TestSuite(std::string path) {
	m_tests.read(path + ".in");
	m_results.read(path + ".expected");
}

void Fuse::TestSuite::run() {
	auto failedCount = 0;
	auto unimplementedCount = 0;
	for (auto test : m_tests.container()) {

		auto key = test.first;
		std::cout << "** Checking: " << key << std::endl;

		auto input = test.second;
		auto result = m_results.container().find(key)->second;

		Fuse::TestRunner runner(input, result);

		runner.run();
		if (runner.failed())
			++failedCount;
		if (runner.unimplemented())
			++unimplementedCount;
	}
	std::cout << "+++ Failed test count: " << failedCount << std::endl;
	std::cout << "+++ Unimplemented test count: " << unimplementedCount << std::endl;
}