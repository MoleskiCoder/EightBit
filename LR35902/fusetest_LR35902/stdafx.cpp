// stdafx.cpp : source file that includes just the standard includes
// fusetest_LR35902.pch will be the pre-compiled header
// stdafx.obj will contain the pre-compiled type information

#include "stdafx.h"

// TODO: reference any additional headers you need in STDAFX.H
// and not in this file
