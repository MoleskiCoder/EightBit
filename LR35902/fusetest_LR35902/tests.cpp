#include "stdafx.h"

#include "FuseTestSuite.h"

int main(int, char*[]) {

	Fuse::TestSuite testSuite("C:\\github\\cpp\\EightBitWrapper\\libraries\\EightBit\\LR35902\\fusetest_LR35902\\fuse-tests\\tests");
	testSuite.run();

	return 0;
}
