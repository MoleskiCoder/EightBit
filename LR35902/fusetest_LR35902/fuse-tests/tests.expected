00
    0 MC 0000
    4 MR 0000 00
0000 0000 0000 0000 0000 0001
0 4

01
    0 MC 0000
    4 MR 0000 01
    4 MC 0001
    7 MR 0001 12
    7 MC 0002
   10 MR 0002 34
0000 3412 0000 0000 0000 0003
0 10
